aten/src/ATen/native/cuda/linalg/BatchLinearAlgebra.cpp

```diff
@@ -2176,12 +2176,29 @@ void svd_magma(const Tensor& A,
                    .mT();
   // U, S, Vh, info are the right size and strides, but are on GPU
   // We copy them into CPU in pinned_memory
-  const auto empty_like_cpu = [](const Tensor& t) {
-    return at::empty_like(t, t.options().device(kCPU).pinned_memory(true));
+  const auto empty_like_cpu = [](const Tensor& t, bool make_contiguous = false) {
+    if (!make_contiguous) {
+      return at::empty_like(t,
+                            t.options()
+                            .device(kCPU)
+                            .pinned_memory(true));
+    } else {
+      return at::empty_like(t.mT(),
+                            t.options()
+                            .device(kCPU)
+                            .memory_format(at::MemoryFormat::Contiguous)
+                            .pinned_memory(true)).mT();
```

Comment:
this is extremely confusing, you are returning a non-contiguous tensor if `make_contiguous` argument is `true`

Reply:
@ngimel Could you please elaborate? The stanza starting at line 2185 is only entered when `make_contiguous` is `true`. As such, we set the memory format to `at::MemoryFormat::Contiguous` on line 2189. Am I misunderstanding how this API works?

Reply:
@alugorey , why `.mT()` calls are needed there?

Reply:
@malfet After investigating and talking amongst the team, we discovered this change to be an artifact left over from an earlier version of development. I have pushed up a new commit removing this unnecessary transposition.

Reply:
And I think this addresses @ngimel's previous comment.